blockchain/test_blockchain.go

```diff
@@ -0,0 +1,273 @@
+package blockchain
+
+import (
+	"encoding/json"
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/NethermindEth/juno/adapters/sn2core"
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/crypto"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/db/pebble"
+	"github.com/NethermindEth/juno/starknet"
+	"github.com/NethermindEth/juno/starknet/compiler"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/stretchr/testify/require"
+)
+
+type testBlockchain struct {
+	Blockchain
+	t                        *testing.T
+	account, deployer, erc20 TestClass
+	addr2classHash           map[felt.Felt]felt.Felt
+}
+
+func NewTestBlockchain(t *testing.T, protocolVersion string) testBlockchain {
+	t.Helper()
+
+	testDB := pebble.NewMemTest(t)
+	chain := testBlockchain{
+		Blockchain:     *New(testDB, &utils.Sepolia),
+		t:              t,
+		addr2classHash: make(map[felt.Felt]felt.Felt),
+	}
+
+	genesisBlock := &core.Block{
+		Header: &core.Header{
+			Number:          0,
+			Timestamp:       0,
+			ProtocolVersion: protocolVersion,
+			Hash:            utils.HexToFelt(t, "0xb00"),
+			ParentHash:      &felt.Zero,
+			GlobalStateRoot: &felt.Zero,
+		},
+	}
+	genesisStateUpdate := &core.StateUpdate{
+		BlockHash: genesisBlock.Hash,
+		NewRoot:   &felt.Zero,
+		OldRoot:   &felt.Zero,
+		StateDiff: &core.StateDiff{},
+	}
+	require.NoError(t, chain.Store(genesisBlock, &core.BlockCommitments{}, genesisStateUpdate, nil))
+
+	const prefix = "../../cairo/scarb/target/dev/"
+	// Predeploy presets
+	// https://github.com/OpenZeppelin/cairo-contracts/tree/main/packages/presets
+	chain.account = NewClass(t, prefix+"juno_AccountUpgradeable.contract_class.json")
+	chain.account.AddAccount(
+		*utils.HexToFelt(t, "0xc01"),
+		*utils.HexToFelt(t, "0x10000000000000000000000000000"),
+	)
+	chain.deployer = NewClass(t, prefix+"juno_UniversalDeployer.contract_class.json")
+	chain.deployer.AddAccount(
+		*utils.HexToFelt(t, "0xc02"),
+		felt.Felt{},
+	)
+	chain.erc20 = NewClass(t, prefix+"juno_ERC20Upgradeable.contract_class.json")
+	chain.erc20.AddAccount(
+		*utils.HexToFelt(t, "0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7"),
+		felt.Felt{},
+	)
+	chain.erc20.AddAccount(
+		*utils.HexToFelt(t, "0x04718f5a0fc34cc1af16a1cdee98ffb20c31f5cd61d6ab07201858f4287c938d"),
+		felt.Felt{},
+	)
```

Comment:
Why not use felt.Zero here instead of felt.Felt. No real difference but a bit a more legible